### PR TITLE
Convert to agda2hs-compatible syntax

### PR DESCRIPTION
**Pull request by flupe**
Split `Scope.agda` into different modules.
I can now generate Haskell with `agda2hs` for files up to `Reduce.agda` and `Syntax.agda` (haven't looked into typing rules yet). The generated code does not compile though, because `agda2hs` doesn't support everything we need.

To be continued.

### CODE REVIEW

File: src/Reduce.agda [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      Nothing  → Nothing
+  step (TApp u es) = fmap (λ u → TApp u es) (step u)
+  step (TPi x a b) = Nothing
+  step (TSort x) = Nothing
+  step (TLet x u v) = case step u of λ where
+    (Just u') → Just (TLet x u' v)
+    Nothing   → Just (substTop (rezz _) u v)
+  {-# COMPILE AGDA2HS step #-}
+
+{-
```
Comment (jespercockx) [inline src/Reduce.agda:100]: This is still commented out! However, I'm anyway working on a new version of reduction so perhaps it's fine.

File: src/Scope.agda [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+open import Utils.Erase
+open import Utils.Tactics
+open import Utils.Dec as Dec
+import Utils.List as List
+
+module Scope where
+
+  ----------------------------
+
+  @0 <>-∅ : {α : Scope name} → α <> empty ≡ α
```
Comment (jespercockx) [inline src/Scope.agda:47]: Uhm I'm confused, how is this the first definition in this module? Should this module even still exist?

File: src/Syntax.agda [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+  coerceEnv (rezz (x ∷ α)) p e =
+    SCons (lookupEnv e _ (bindSubToIn p))
+          (coerceEnv (rezz α) (joinSubRight (rezz [ get x ]) p) e)
+  {-# COMPILE AGDA2HS coerceEnv #-}
+
+  dropEnv : {@0 α β : Scope name} {@0 x : name} → (x ◃ α) ⇒ β → α ⇒ β
+  dropEnv (SCons x f) = f
+  {-# COMPILE AGDA2HS dropEnv #-}
+
+-- TODO(flupe): move this out
```
Comment (jespercockx) [inline src/Syntax.agda:180]: TODO reminder

File: src/Utils.agda [hunk 4]
```diff
@@ -3,32 +3,20 @@
 module Utils where
```
Comment (jespercockx) [inline src/Utils.agda:3]: Maybe this module should be removed since it's been split up? Or just become a list of re-exports of all the `Utils.*` modules?

**Review by jespercockx — COMMENTED** [review]
Thank you, this is great! I have a few comments before merging it though.

File: src/Scope.agda (hunk 2 above)
Comment (flupe) [reply src/Scope.agda:47]: This module contains what I haven't yet moved to somewhere else in `Scope/`. It no longer typechecks and will get removed.

File: lib/Reduce.hs [hunk 5]
```diff
@@ -0,0 +1,106 @@
+{-# LANGUAGE ScopedTypeVariables #-}
```
[the remark this answers is not in the document]
Comment (flupe) [reply lib/Reduce.hs:1]: Will remove.

File: src/Reduce.agda (hunk 1 above)
Comment (flupe) [reply src/Reduce.agda:100]: Surprisingly hard to work a *terminating* fueled function without pattern-matching on `Nat` using the prelude...

Comment (jespercockx) [reply src/Reduce.agda:100]: This is a good point and probably something we should think about. I think for now the best option is to just leave the fuel in the generated Haskell code as an `Int`.

**Comment by jespercockx** [comment]
I'm merging this now, feel free to create a new PR if you continue working on this.

**Pull request merged**